### PR TITLE
add README.md for Sentence transformer examples with HPU device

### PR DESCRIPTION
**Pull request by ZhengHongming888**
# What does this PR do?

This PR will add one README.md for all sentence transformer HPU examples although they have each readme on how to run for each example. This README will give the whole pictures on all examples for sentence transformer with HPU device and their supported features/test configurations as below like :

image

## Before submitting
- [X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: examples/sentence-transformers-training/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Examples for Sentence Transformer
+
+We provide 3 examples to show how to use the sentence transformer with HPU devices. 
+
+- **[training_stsbenchmark.py](https://github.com/huggingface/optimum-habana/tree/main/examples/sentence-transformers-training/sts)** - This example shows how to create a SentenceTransformer model from scratch by using a pre-trained transformer model (e.g. [`distilbert-base-uncased`](https://huggingface.co/distilbert/distilbert-base-uncased)) together with a pooling layer.
```
Comment (hchauhan123) [inline examples/sentence-transformers-training/README.md:5]: *Sentence Transformers

File: examples/sentence-transformers-training/README.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+- **[training_nli.py](https://github.com/huggingface/optimum-habana/tree/main/examples/sentence-transformers-training/nli)** - This example gives two sentences (premise and hypothesis) and the task of Natural Language Inference (NLI) is to decide if the premise entails the hypothesis, if they are contradiction, or if they are neutral. Commonly the NLI dataset in [SNLI](https://huggingface.co/datasets/stanfordnlp/snli) and [MultiNLI](https://huggingface.co/datasets/nyu-mll/multi_nli) are used.
```
Comment (hchauhan123) [inline examples/sentence-transformers-training/README.md:7]:
```suggestion
- **[training_nli.py](https://github.com/huggingface/optimum-habana/tree/main/examples/sentence-transformers-training/nli)** - This example provides two sentences (a premise and a hypothesis), and the task of Natural Language Inference (NLI) is to determine whether the premise entails the hypothesis, contradicts it, or if they are neutral. Commonly the NLI dataset in [SNLI](https://huggingface.co/datasets/stanfordnlp/snli) and [MultiNLI](https://huggingface.co/datasets/nyu-mll/multi_nli) are used.
```

**Review by hchauhan123 — CHANGES_REQUESTED** [review]

File: examples/sentence-transformers-training/README.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# Examples for Sentence Transformers
+
+We provide 3 examples to show how to use the Sentence Transformers with HPU devices.
```
Comment (libinta) [inline examples/sentence-transformers-training/README.md:3]: Gaudi

File: examples/sentence-transformers-training/README.md (hunk 1 above)
Comment (ZhengHongming888) [reply examples/sentence-transformers-training/README.md]: done!

**Review by hchauhan123 — APPROVED** [review]

**Review by regisss — APPROVED** [review]

**Pull request merged**